src/core/issues-notes.js

```diff
@@ -146,6 +159,36 @@ function handleIssues($ins, ghIssues, conf) {
   }
 }
 
+//derives the text-color-class based on the illumination score of the labelColor
+function deriveTextColorClass(hexColor) {
+  try {
+    const rgb = parseInt(hexColor, 16);
+    if (isNaN(rgb)) {
+      throw NaN;
```

Comment:
ProTip: never throw anything apart from an actual Error or something that inherits from Error (e.g., `TypeError`). Here we want a type error... 

Also, making the whole body of the function a try/catch is not a good idea. We don't want to mask programming errors, with input errors.  

Thus, this should just be:

```JS
function deriveTextColorClass(hexColor) {
  const rgb = parseInt(hexColor, 16);
  if (isNaN(rgb)) {
    throw new TypeError(`Invalid hex color: ${hexColor}`);
  }
  const red = (rgb >> 16) & 0xff;
  const green = (rgb >> 8) & 0xff;
  const blue = (rgb >> 0) & 0xff;
  const illumination = 0.2126 * red + 0.7152 * green + 0.0722 * blue;
  return illumination > 140 ? "light" : "dark";
}
```

Reply:
to be clear, you always want `Error`, because otherwise you don't get a stack trace.